src/views/dashboard/Dashboard.tsx

```diff
@@ -8,18 +8,13 @@ import Button from "@material-ui/core/Button";
 const Dashboard = (props: any) => {
    //const classes = useStyles();
    const [name, setName] = useState("Anonymous");
-   const [user_id, setUser] = useState("");
 
    useEffect(() => {
       let username: string | null = localStorage.getItem("username");
       if (username) {
          setName(username);
       }
-      let user_id: string | null = localStorage.getItem("user_id");
-      if (user_id) {
-         setUser(user_id);
-      }
-   }, []);
+   }, [name]);
```

Comment:
Will this cause an infinite loop?

Reply:
I believe it will cause an infinite loop because you're changing the name as part of the loop which will cause your dependency array to be triggered.